torch/optim/optimizer.py

```diff
@@ -34,6 +35,24 @@ def __init__(self, params, defaults):
         torch._C._log_api_usage_once("python.optimizer")
         self.defaults = defaults
 
+        self._zero_grad_profile_name = "{}#{}.{}".format("Optimizer.zero_grad", self.__class__.__name__, "zero_grad")
```

Comment:
not sure why zero_grad fails with an AttributeError

@vincentqb @VitalyFedyunin @dzhulgakov - any idea what could happen here?

Reply:
@ilia-cher What's the calling context of this error? 
It seems a new Adam object is created by deepcopy and not goes through __init__. Because Optimizer's "__getstate__" doesn't return "self._zero_grad_profile_name", the new created object fails to have this attribute.
Should we support profiling on the objects that created by deepcopy? If so, more code needs to be changed.

Reply:
That's right, in this case optimizer is used in the multiprocessing context, so it gets picked and unpickled.